Option for ignoring the Content-Range header

Stadly opened the pull request:
<!--
Fill in the relevant information below to help triage your issue.

Pick the target branch based on the following criteria:
  * Documentation improvement: master branch
  * Bugfix: master branch
  * QA improvement (additional tests, CS fixes, etc.) that does not change code
    behavior: master branch
  * New feature, or refactor of existing code: develop branch

You MUST provide a signoff in your commits for us to be able to accept your
patch; you can do this by providing either the --signoff or -s flag when using
"git commit". Please see the project contributing guide and
https://developercertificate.org for details.
-->

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

<!--
Tell us about why this change is necessary:
- Are you fixing a bug or providing a failing unit test to demonstrate a bug?
  - How do you reproduce it?
  - What did you expect to happen?
  - What actually happened?
  - TARGET THE master BRANCH

- Are you adding documentation?
  - TARGET THE master BRANCH

- Are you providing a QA improvement (additional tests, CS fixes, etc.) that
  does not change behavior?
  - Explain why the changes are necessary
  - TARGET THE master BRANCH

- Are you fixing a BC Break?
  - How do you reproduce it?
  - What was the previous behavior?
  - What is the current behavior?
  - TARGET THE master BRANCH

- Are you adding something the library currently does not support?
  - Why should it be added?
  - What will it enable?
  - How will the code be used?
  - TARGET THE develop BRANCH

- Are you refactoring code?
  - Why do you feel the refactor is necessary?
  - What types of refactoring are you doing?
  - TARGET THE develop BRANCH
-->

When the response body has already been limited to the requested range, the emitter should ignore the Content-Range header and emit the full response body.

Fixes #22

samsonasik commented on test/Emitter/SapiStreamEmitterTest.php:
```diff
@@ -35,10 +35,14 @@
 class SapiStreamEmitterTest extends AbstractEmitterTest
 {
+    /** @var SapiStreamEmitter */
+    protected $emitterIgnoreContentRange;
```
private modifier should be fine since this class is not to be extended.

```suggestion
    private $emitterIgnoreContentRange;
```

lcobucci commented:
Based on the discussion in #22, I'd say it makes more sense to provide an alternative implementation instead of modifying the existing one.

I'd be fine with reorganising `SapiStreamEmitter` to provide some decoration but having a different behaviour based on a boolean constructor parameter isn't a very maintainable/extensible design IMHO.

Stadly committed:
Option for ignoring the Content-Range header

When the response body has already been limited to the requested range, the emitter should ignore the Content-Range header and emit the full response body.

Signed-off-by: Stadly <magnar@myrtveit.com>

Stadly commented:
Any other opinions? I'm fine with either:

1. Modify `SapiStreamEmitter` (as done in this PR).
2. Create a separate emitter.
3. Fix the memory issue in `SapiEmitter`. That would also fix #22, as `Content-Range` is already ignored by `SapiEmitter`. I think this is my preferred option. Or is there a reason why `SapiEmitter` converts the whole response body to a string and emits it in a single go instead of emitting it as smaller chunks? With the current implementation of `SapiEmitter`, the entire response body must be written to memory, which does not work for large responses.
